src/libs/services/NetService.ts

```diff
@@ -38,33 +39,36 @@ export const useNetService = (client?: CozyClient): void =>
 const waitForOnline = (callbackRoute: string): void => {
   log.debug('Adding NetInfo listener')
 
-  const callback = (state: NetInfoState): void => {
+  // Define the unsubscribe function inside the listener
+  const unsubscribe = NetInfo.addEventListener(state => {
     if (state.isConnected) {
-      log.debug('Removing NetInfo listener')
+      log.debug('NetService is online, navigating to callback route')
 
       callbackRoute === routes.stack && showSplashScreen().catch(log.error)
 
       try {
-        reset(callbackRoute)
+        navigate(callbackRoute)
       } catch (error) {
         log.error(error)
       }
 
+      log.debug('NetService redirect done, removing NetInfo listener')
+      // Unsubscribe from the NetInfo event listener, as we're now online
```

Comment:
nit: seems like this comment is redundant since we already have a log explaining what's happening?